src/osd/OSD.cc

```diff
@@ -759,8 +759,8 @@ void OSDService::update_osd_stat(vector<int>& hb_peers)
 {
   Mutex::Locker lock(stat_lock);
 
-  osd_stat.hb_in.swap(hb_peers);
-  osd_stat.hb_out.clear();
+  osd_stat.hb_out.swap(hb_peers);
```

Comment:
@tchaikov. First, it is correct, update_osd_stat() is only called by OSD::heartbeat, and hb_peers is an input parameter for lupdate_osd_stat. 

Let's take a look how is "hb_peers" used in OSD::heartbeat: "hb_peers" is got from heartbeat_peers, and heart beat will be sent to these peers one by one from current osd by using send_message.

In this way, these hb_peers are hb_out, not hb_in. That is what I want to fix in this PR.

Then for the issue 2896. As I described above, "hb_out" is what we want to display, and "hb_in" is vestigal.